### DISCUSSION TITLE
How to access this cluster from another container running memtier_benchmark?

### DISCUSSION THREAD

**Discussion by TobyHFerguson** [Grokzen/docker-redis-cluster#131] [Q&A] [answered]
I want to use two containers, one running this cluster, the other running a tool such as memtier_benchmark, or just the redis-cli. However this doesn't work because this cluster always redirects to 127.0.0.1 as its address, and the other container can't figure out the true address needed!

Is there a way to set the IP address this cluster uses to its address allocated by docker?

**Comment by Grokzen** [comment]
First i would not really run any benchmarks or anything similar to get an accurate reading or view of how performant redis cluster really is in a single container solution. You should really build your dev/staging/prod envrionment and run the tests against those kinds of builds/setup instead of doing it against this container. This container and setup is not really or ever will be really suited for production environment or similar setups where performance is really a factor or a consideration.

But... to answer your question, the only suggestion i have is to attempt to setup both containers with a host network so that the inside of both containers is exposed to 127.0.0.1 on both ends, then any tool in any other container should work out fine i guess. If you get that to work, please drop a line as it might be good information for future us to see :)

**Comment by TobyHFerguson** [comment, marked as answer]
Thanks! Actually I’m not doing any benchmarking at all! I’m simply using memtier_benchmark as a client to put some data in the system. My real goal is to use the subsequent cluster as a platform against which to test a metrics gathering client. 

I solved the problem:

My script builds a network with a known subnet, allocates a specific address to the cluster and then (after sleeping a little), connects from the memtier_benchmark container. 

```
NET=redisClusterNet
RC=redis_cluster
MC=memtier_benchmark
SUBNET=192.168.0.0/16
GATEWAY=192.168.0.1
IP=192.168.0.10
PORT=7000
docker network create --driver=bridge --subnet=${SUBNET:?} --gateway=${GATEWAY:?} ${NET:?} 
docker run --rm -d --name ${RC:?} -e IP --ip ${IP:?} --net ${NET:?} -p 7000-7005:7000-7005 grokzen/redis-cluster:latest
sleep 5
docker run --name ${MC:?} --rm -it --net ${NET:?} redislabs/memtier_benchmark -s ${IP:?} -p ${PORT:?} --cluster-mode
```

Hope that helps in your future efforts!

**Reply by Grokzen** [reply]
Cool that you found a solution :) I would still probably have gone with a more basic host network to avoid having to create a new network and use other IP addresses but that looks like it works out well.
